plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsReader.java

```diff
@@ -176,23 +206,115 @@ else if (columnHandle.getBaseType() == VARBINARY) {
+                    // Fail loud upon duplicates (there must be none)
+                    .collect(toImmutableMap(blobMetadata -> getOnlyElement(blobMetadata.fields()), identity()));
+
+            for (Entry<Integer, BlobMetadata> entry : thetaBlobsByFieldId.entrySet()) {
+                int fieldId = entry.getKey();
+                BlobMetadata blobMetadata = entry.getValue();
+                String ndv = blobMetadata.properties().get(APACHE_DATASKETCHES_THETA_V1_NDV_PROPERTY);
+                if (ndv == null) {
+                    log.debug("Blob %s is missing %s property", blobMetadata.type(), APACHE_DATASKETCHES_THETA_V1_NDV_PROPERTY);
+                    remainingColumnIds.remove(fieldId);
+                }
+                else {
+                    remainingColumnIds.remove(fieldId);
+                    ndvByColumnId.put(fieldId, parseLong(ndv));
+                }
+            }
+        }
+
+        // TODO (https://github.com/trinodb/trino/issues/15397): remove support for Trino-specific statistics properties
+        Iterator<Entry<String, String>> properties = icebergTable.properties().entrySet().iterator();
+        while (!remainingColumnIds.isEmpty() && properties.hasNext()) {
+            Entry<String, String> entry = properties.next();
+            String key = entry.getKey();
+            String value = entry.getValue();
             if (key.startsWith(TRINO_STATS_PREFIX)) {
                 Matcher matcher = TRINO_STATS_NDV_PATTERN.matcher(key);
                 if (matcher.matches()) {
                     int columnId = Integer.parseInt(matcher.group("columnId"));
-                    long ndv = Long.parseLong(value);
-                    ndvByColumnId.put(columnId, ndv);
+                    if (remainingColumnIds.remove(columnId)) {
+                        long ndv = parseLong(value);
+                        ndvByColumnId.put(columnId, ndv);
+                    }
                 }
             }
-        });
+        }
+
         return ndvByColumnId.buildOrThrow();
     }
+
+    /**
+     * Iterates over existing statistics files present for parent snapshot chain,  starting at {@code startingSnapshotId} (inclusive).
+     */
+    public static Iterator<StatisticsFile> walkStatisticsFiles(Table icebergTable, long startingSnapshotId)
+    {
+        return new AbstractIterator<>()
+        {
+            private final Map<Long, StatisticsFile> statsFileBySnapshot = icebergTable.statisticsFiles().stream()
+                    .collect(toMap(
+                            StatisticsFile::snapshotId,
+                            identity(),
+                            (a, b) -> {
+                                throw new IllegalStateException("Unexpected duplicate statistics files %s, %s".formatted(a, b));
+                            },
+                            HashMap::new));
+
+            private final Iterator<Long> snapshots = walkSnapshots(icebergTable, startingSnapshotId);
```

Comment:
`snapshots` -> `parentSnapshots` ?

Reply:
"parent Snapshots" would not include the starting point (i am not a parent of myself)